src/derivatives.jl

```diff
@@ -0,0 +1,162 @@
+function derivative(A::LinearInterpolation{<:AbstractVector{<:Number}}, t::Number)
+    idx = findfirst(x -> x >= t, A.t) - 1
```

Comment:
If we assume monotonicity, then this can be
```suggestion
    idx = searchsortedfirst(A.t, t) - 1
```